demos/cipher/aesgcm.c

```diff
@@ -122,6 +123,19 @@ static int aes_gcm_encrypt(void)
     if (!EVP_EncryptFinal_ex(ctx, outbuf, &tmplen))
         goto err;
 
+    /* Retrieve service indicator for IV generation */
+    iv_gen_rand = -1;
+    params[0] = OSSL_PARAM_construct_int(OSSL_CIPHER_PARAM_AEAD_IV_GENERATED, &iv_gen_rand);
+    if (!EVP_CIPHER_CTX_get_params(ctx, params))
+        goto err;
+    printf("Service indicator for OSSL_CIPHER_PARAM_AEAD_IV_GENERATED is: %d\n", iv_gen_rand);
```

Comment:
I'd wrapper this in a call to `OSSL_PARAM_modified` to account for providers that don't know about the parameter.

Reply:
OOOOOOH I didn't know that API, will play with it.

Reply:
The other useful one is `OSSL_PARAM_set_all_unmodified` to reset the modified flagging for a param array to all reuse.